### PR TITLE
Signup view model tests

### PR DESCRIPTION
**Pull request by danielaRiesgo**
## Sumary ##
**Trello card**: https://trello.com/c/FJ6N5DPv/33-hacer-tests-del-signupviewmodel

Adding sign up view model tests.
Refactoring login view model tests when refactoring session service mock for tests to test view model and not mock logic.
Adding toggle visibility tests to login view model ones.

### CODE REVIEW

**Comment by guidomb** [comment]
👍

File: AuthenticationTests/LogIn/LoginViewModelTests.swift [hunk 1]
```diff
@@ -18,12 +18,12 @@ class LoginViewModelSpec: QuickSpec {
     
     override func spec() {
         describe("LoginViewModel") {
-            
+
             var sessionService: MockSessionService!
             var loginVM: LoginViewModel<MyUser, MockSessionService>!
```
Comment (FranDepascuali) [inline AuthenticationTests/LogIn/LoginViewModelTests.swift]: this should be `loginViewModel`

File: AuthenticationTests/LogIn/LoginViewModelTests.swift [hunk 2]
```diff
@@ -36,12 +36,36 @@ class LoginViewModelSpec: QuickSpec {
                 expect(loginVM.passwordVisible.value) == false
             }
             
+            describe("#togglePasswordVisibility") {
+                
+                context("when executing action") {
+                    
+                    it("should change #passwordVisible from false to true") { waitUntil { done in
```
Comment (FranDepascuali) [inline AuthenticationTests/LogIn/LoginViewModelTests.swift]: I would move the false and true part to the context:
```swift
context("When executing action") {
    context("When #passwordVisible is false") {
        it("should change it to true") {
        }
    }
    context("When #passwordVisible is true") {
        it("should change it to false") {
        }
    }
}
```
However, you decide, it isn't so important so if you want to leave it like this its okay

**Comment by FranDepascuali** [comment]
👍

**Pull request merged**